### ISSUE TITLE
dictionary does not work

### ISSUE THREAD

**Issue by enigmasc** [koreader/koreader#11810]
* KOReader version: koreader-kobo-v2024.04-48-g3e0418463_2024-05-13
* Device: Kobo Touch (N905C)

#### Issue

Dictionary stops at "Searching......." without getting any definitions. When quitting Koreader, it crashes and then restarts.
[crash.zip](https://github.com/koreader/koreader/files/15291983/crash.zip)

![Reader_2024-05-13_160821](https://github.com/koreader/koreader/assets/74174080/8d201699-05ad-4b33-980d-ecfa5000cfac)

**Comment by hius07** [comment]
Pinging @comphilip

**Comment by Frenzie** [comment]
I don't think the Kobo Touch is a non-touch. ;-)

**Comment by hius07** [comment]
https://github.com/koreader/koreader/blob/3e04184638df5dc92986d821af172a545654c3b4/frontend/ui/widget/buttontable.lua#L153-L154

**Comment by comphilip** [comment]
@hius07 I check nil for all variables, but I missed ButtonTable.layout is not always available.

**Comment by enigmasc** [comment]
The dictionary bug is not fixed and the new update (koreader-kobo-v2024.04-63-gfd5260f2c_2024-05-1) makes the problem worse. Koreader just freezes on my Kobo Touch now. Have to roll back to koreader-kobo-v2024.04-34-gbaab32633_2024-05-11.
[Koreader Crash.zip](https://github.com/koreader/koreader/files/15319985/Koreader.Crash.zip)

**Comment by ptrm** [comment]
Same, Clara BW, in case another log is needed:

```
05/15/24-12:31:22 INFO  Inhibiting user input
05/15/24-12:31:23 INFO  Loading plugins from directory: plugins
05/15/24-12:31:24 INFO  Restoring user input handling
g_mkdir failed: No such file or directory
05/15/24-12:31:29 WARN  error in wrapped function: frontend/ui/widget/dictquicklookup.lua:724: bad argument #1 to 'insert' (table expected, got nil)
stack traceback:
        [C]: in function 'insert'
        frontend/ui/widget/dictquicklookup.lua:724: in function 'init'
        frontend/ui/widget/widget.lua:46: in function 'new'
        frontend/apps/reader/modules/readerdictionary.lua:1001: in function 'showDict'
        frontend/apps/reader/modules/readerdictionary.lua:995: in function 'stardictLookup'
        frontend/apps/reader/modules/readerdictionary.lua:443: in function <frontend/apps/reader/modules/readerdictionary.lua:442>
        [C]: in function 'xpcall'
        frontend/ui/trapper.lua:48: in function <frontend/ui/trapper.lua:45>
g_mkdir failed: No such file or directory
05/15/24-12:31:38 WARN  error in wrapped function: frontend/ui/widget/dictquicklookup.lua:724: bad argument #1 to 'insert' (table expected, got nil)
stack traceback:
        [C]: in function 'insert'
        frontend/ui/widget/dictquicklookup.lua:724: in function 'init'
        frontend/ui/widget/widget.lua:46: in function 'new'
        frontend/apps/reader/modules/readerdictionary.lua:1001: in function 'showDict'
        frontend/apps/reader/modules/readerdictionary.lua:995: in function 'stardictLookup'
        frontend/apps/reader/modules/readerdictionary.lua:443: in function <frontend/apps/reader/modules/readerdictionary.lua:442>
        [C]: in function 'xpcall'
        frontend/ui/trapper.lua:48: in function <frontend/ui/trapper.lua:45>
g_mkdir failed: No such file or directory
05/15/24-12:31:44 WARN  error in wrapped function: frontend/ui/widget/dictquicklookup.lua:724: bad argument #1 to 'insert' (table expected, got nil)
stack traceback:
        [C]: in function 'insert'
        frontend/ui/widget/dictquicklookup.lua:724: in function 'init'
        frontend/ui/widget/widget.lua:46: in function 'new'
        frontend/apps/reader/modules/readerdictionary.lua:1001: in function 'showDict'
        frontend/apps/reader/modules/readerdictionary.lua:995: in function 'stardictLookup'
        frontend/apps/reader/modules/readerdictionary.lua:443: in function <frontend/apps/reader/modules/readerdictionary.lua:442>
        [C]: in function 'xpcall'
        frontend/ui/trapper.lua:48: in function <frontend/ui/trapper.lua:45>
```

**Comment by hius07** [comment]
The fix is not merged yet: https://github.com/koreader/koreader/pull/11811.

**Comment by hius07** [comment]
> Koreader just freezes on my Kobo Touch now

Not related. Reported in https://github.com/koreader/koreader/issues/11824.

**Comment by ptrm** [comment]
> The fix is not merged yet: #11811.

@hius07 replied too hastily, sorry ;) Replaced the fixed file manually and it works, thanks!
